src/custom_nodes/east_ocr/east_ocr.cpp

```diff
@@ -0,0 +1,399 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "../../custom_node_interface.h"
+#include "nms.hpp"
+#include "opencv2/opencv.hpp"
+#include "utils.hpp"
+
+static constexpr const char* IMAGE_TENSOR_NAME = "image";
+static constexpr const char* SCORES_TENSOR_NAME = "scores";
+static constexpr const char* GEOMETRY_TENSOR_NAME = "geometry";
+static constexpr const char* TEXT_IMAGES_TENSOR_NAME = "text_images";
+static constexpr const char* COORDINATES_TENSOR_NAME = "text_coordinates";
+static constexpr const char* CONFIDENCE_TENSOR_NAME = "confidence_levels";
+
+int extract_text_images_into_output(struct CustomNodeTensor* output, const std::vector<cv::Rect>& boxes, const cv::Mat& originalImage, int targetImageHeight, int targetImageWidth, bool convertToGrayScale, const std::string& imageLayout) {
```

Comment:
Maybe move to utils as well?

Reply:
This is very specific to this custom node and will not be reused in other nodes. Utils contain reusable components.